static/js/util/profile_edit.js

```diff
@@ -334,72 +300,94 @@ export function boundMonthYearField(keySet, label) {
     let formattedDate = _.get(profile, keySet);
 
     if (formattedDate !== undefined && formattedDate !== null) {
-      return moment(formattedDate, DATE_FORMAT);
+      return moment(formattedDate, ISO_8601_FORMAT);
     }
     return null;
   };
 
-  // Get a tuple { month, year } which contains the values being edited in the textbox
+  // Get a tuple { date, month, year } which contains the values being edited in the textbox
   // values may be strings or numbers. Otherwise return empty object.
-  let getMonthYear = () => {
-    let monthYear = _.get(profile, editKeySet, {});
+  let getTuple = () => {
+    let tuple = _.get(profile, editKeySet, {});
 
-    if (_.isEmpty(monthYear)) {
+    if (_.isEmpty(tuple)) {
       let date = getDate();
       if (date !== null && date.isValid()) {
         return {
           month: date.month() + 1,
-          year: date.year()
+          year: date.year(),
+          date: date.date()
         };
       }
     }
-    return monthYear;
+    return tuple;
```

Comment:
it looks like we're dealing with an object ( `{ month: ..., year: ..., date: ...}` ) here, which makes me feel like `tuple` is a bit of a confusing name (JS doesn't have tuples, after all).